driver-sync/src/test/functional/com/mongodb/client/AbstractSessionsProseTest.java

```diff
@@ -103,4 +113,79 @@ public void commandStarted(final CommandStartedEvent event) {
             assertEquals(1, minLsidSetSize);
         }
     }
+
+    @Test
+    public void shouldIgnoreImplicitSessionIfConnectionDoesNotSupportSessions() throws IOException {
+        assumeTrue(serverVersionAtLeast(4, 2));
+        Process mongocryptdProcess = startMongocryptdProcess("1");
+        try {
+            AtomicBoolean containsLsid = new AtomicBoolean(true);
+            try (MongoClient client = getMongoClient(
+                    getMongocryptdMongoClientSettingsBuilder()
+                            .addCommandListener(new CommandListener() {
+                                @Override
+                                public void commandStarted(final CommandStartedEvent event) {
+                                    containsLsid.set(event.getCommand().containsKey("lsid"));
+                                }
+                            })
+                            .build())) {
+
+                MongoCollection<Document> collection = client.getDatabase(getDefaultDatabaseName()).getCollection(getClass().getName());
+                try {
+                    collection.find().first();
+                } catch (MongoCommandException e) {
+                    // ignore command errors from mongocryptd
+                }
+                assertFalse(containsLsid.get());
+            }
+        } finally {
+            mongocryptdProcess.destroy();
+        }
+    }
+
+    @Test
+    public void shouldThrowOnExplicitSessionIfConnectionDoesNotSupportSessions() throws IOException {
```

Comment:
In this case I couldn't find any other prose tests for the sessions spec actually implemented so I added them to the existing file.  I wouldn't want to split the two new ones since they seem so similar

Reply:
Fair enough